src/graph/network/communicator.h

```diff
@@ -16,7 +16,7 @@ namespace network {
  *
  * Sender is an abstract class that defines a set of APIs for sending 
  * binary data over network. It can be implemented by different underlying 
- * networking libraries such TCP socket and ZMQ. One Sender can connect to 
+ * networking libraries such TCP socket and MPI. One Sender can connect to 
```

Comment:
Why do we prefer MPI over ZMQ?

Reply:
> Why do we prefer MPI over ZMQ?

MPI can support GPU-direct communication easily.